[Work-in-Progress] Allow adding a Certificate to a Sequence

Deep-Chill opened the pull request:
- Updated logic for checking progress completion
- Updated sequences page to display each EnumeratedElement while accounting for certificates and live events.
- Added properties on EnumeratedElements to indicate LiveEvent or Certificate status.

Notes:
- The last element of a sequence is considered its certificate
- The API view simply sets has_certificate to True, then the user has to add a PageElement with the highest rank for it to be the certificate.

Deep-Chill committed:
Allow adding a Certificate to a Sequence

- Added API endpoint in the Sequence viewset to allow adding a certificate to a Sequence
- Updated existing code for the CertificateDownlodView
- Removed Certificate model
- Added a has_certificate field to the Sequence model.

Deep-Chill committed:
Minor update

Deep-Chill committed:
Merge branch 'master' into Certificate-v3

Deep-Chill committed:
Code refactor

- Updated logic for checking progress completion
- Updated sequences page to display each EnumeratedElement while accounting for certificates and live events.
- Added properties on EnumeratedElements to indicate LiveEvent or Certificate status.

Notes:
- The last element of a sequence is considered its certificate
- The API view simply sets has_certificate to True, then the user has to add a PageElement with the highest rank for it to be the certificate.

Deep-Chill committed:
Add new fixtures

3 new fixtures:
- Without a certificate
- With a certificate
- With a certificate and live event

Deep-Chill committed:
Minor update

Deep-Chill committed:
Minor changes

smirolo commented on pages/templates/pages/app/sequences/index.html:
```diff
@@ -10,21 +10,37 @@ <h2>{{ sequence.title }}</h2>
-                        <a href="{% url 'sequences:sequence_page_element_view' sequence=element.sequence.slug rank=element.rank %}">
+                        {% if element.is_live_event %}
+                            <span class="live-event-label">
+                                <a href="{% url 'sequences:sequence_page_element_view' sequence=element.sequence.slug rank=element.rank %}">
```
Typically we pass the URLs in an `urls` variable through the `context` such as to make the templates Jinja2 compatible.

smirolo reviewed (approved):
URLs in the HTML templates will need to be changed. Otherwise let's try it. Thank you!

Deep-Chill committed:
Updated logic

- Updated logic for adding a certificate to a Sequence by moving the logic to the add_element action. Added a write_only certificate field for the element. Added has_certificate field to match the Sequence model.

- Updated front-end views and templates with clearer code, and to use URLs from the context.

smirolo commented on pages/models.py:
```diff
@@ -416,7 +416,7 @@ def __str__(self):
     def get_certificate(self):
         if self.has_certificate:
-            return self.sequence_enumerated_elements.order_by('rank').last()
+            return self.sequence_enumerated_elements.order_by('rank').last().page_element
```
It would be interesting to see the SQL generated by Django ORM for this query vs.
`self.sequence_enumerated_elements.order_by('-rank').first()`

smirolo reviewed (approved):
Looks good to start testing. Thank you!
